### ISSUE TITLE
Lib throws when encounters a Git repo's pubspec.yaml without a version

### ISSUE THREAD

**Issue by sethladd** [dart-lang/pub_cache#17]
Yes, apparently it can happen. I had a pubspec.yaml file in my .pub-cache/git that did not have a version.

It was async_await:

```
name: async_await
description: Compile-time support for "async/await" syntax in Dart.
dependencies:
  analyzer: '>=0.22.4 <0.23.0'
  barback: '>=0.15.0 <0.16.0'
dev_dependencies:
  http: any
  unittest: any
```

Which throws this error:

```
Uncaught Error: Invalid argument(s)
Stack Trace:
#0      _JSSyntaxRegExp.firstMatch (dart:core-patch/regexp_patch.dart:142)
#1      Version.Version.parse (package:pub_semver/src/version.dart:111)
#2      GitDirectoryPackageRef.GitDirectoryPackageRef (package:pub_cache/src/impl.dart:119)
#3      PubCache._parse.<anonymous closure> (package:pub_cache/pub_cache.dart:121)
#4      MappedIterator.moveNext (dart:_internal/iterable.dart:390)
#5      List.addAll (dart:core-patch/growable_array.dart:183)
#6      PubCache._parse (package:pub_cache/pub_cache.dart:122)
#7      PubCache.PubCache (package:pub_cache/pub_cache.dart:41)
#8      _doLoadWhenSnapshot (package:dartdoc/resource_loader.dart:38)
#9      _doLoad (package:dartdoc/resource_loader.dart:24)
#10     loadAsString (package:dartdoc/resource_loader.dart:16)
#11     main.<main_async_body> (file:///Users/sethladd/Code/dartdoc/bin/dart_resource_loader.dart:4)
#12     Future.Future.microtask.<anonymous closure> (dart:async/future.dart:144)
#13     _asyncRunCallbackLoop (dart:async/schedule_microtask.dart:41)
#14     _asyncRunCallback (dart:async/schedule_microtask.dart:48)
#15     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:96)
#16     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:143)

#0      _rootHandleUncaughtError.<anonymous closure> (dart:async/zone.dart:886)
#1      _asyncRunCallbackLoop (dart:async/schedule_microtask.dart:41)
#2      _asyncRunCallback (dart:async/schedule_microtask.dart:48)
#3      _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:96)
#4      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:143)
```

Suggest we tolerate pubspec's without versions in them.

**Comment by sethladd** [comment]
This might do it: https://github.com/dart-lang/pub_cache/pull/18

**Comment by devoncarew** [comment]
pushed the fix as `0.0.1+1`
